python/cudf/cudf/core/series.py

```diff
@@ -453,31 +452,20 @@ def empty(self):
         return not len(self)
 
     def __getitem__(self, arg):
-        data = self._column[arg]
-        index = self.index.take(arg)
-        if is_scalar(data) or data is None:
-            return data
-        return self._copy_construct(data=data, index=index)
-
-    def __setitem__(self, key, value):
-        # coerce value into a scalar or column
-        if is_scalar(value):
-            value = to_cudf_compatible_scalar(value)
+        if isinstance(arg, (slice, range, ColumnBase)) or (
+            pd.api.types.is_bool_dtype(as_column(arg).dtype)
+        ):
+            return self.iloc[arg]
```

Comment:
If slicing via `__getitem__` shouldn't it still slice the index as opposed to by rows?

Reply:
What if you have something like `Series([1, 2, 3], index=[2, 1, 3])`, in this case slice(2:3) would be
```
>>> k[2:3]
3    3
```

I saw one case where index is string type, where index slicing and row slicing both works
```
>>> k=pd.Series([1, 2, 3], index=["a",  "b", "c"])
>>> k[2:3]
c    3
dtype: int64
>>> k["a":"b"]
a    1
b    2

```

Reply:
I'm having trouble following what the expected behaviors for slicing with integers are. Does it always select by row number or index value?

Reply:
by row for slicing when we have integers